### ISSUE TITLE
Can't send smtp mail to smtp tls1.2 only [OpenSSL]

### ISSUE THREAD

**Issue by dnsBlah** [PHPMailer/PHPMailer#2192]
Hi all,

I'm out of thoughts and think this is the best place to ask for support.
There is a smtp server which only accepts TLS1.2 protocol, and the server does not provide STARTTLS

PHPMailer just connects, and the server is accepting connection but not responding resulting the PHP script to run infinite and/or timeout.

**Comment by Synchro** [comment]
There's not enough info to go on here. You need to show your debug output with `SMTPDebug = 3`. Please try the tests using `openssl` described in the troubleshooting guide. To test it further you can use tools like [testssl.sh](https://testssl.sh) which will give you a much more thorough breakdown of what your mail server supports with TLS.

**Comment by dnsBlah** [comment]
Hi @Synchro ,

Thanks for your time!

I'm not sure what to test using testssl.sh
Please advice how to continue.


Below some more information:
Without TLS set:
Executing plain script::
```
2020-11-10 12:22:07	Connection: opening to smtp.mycompany.com:25, timeout=25, options=array()
2020-11-10 12:22:07	Connection: opened
^C
^ Hang here

```

Executing script with TLS1.2 set:
```
2020-11-10 12:22:40	Connection: opening to smtp.mycompany.com:25, timeout=25, options=array (
                   	                    'ssl' => 
                   	                    array (
                   	                      'crypto_method' => 33,
                   	                    ),
                   	                  )
2020-11-10 12:22:40	Connection: opened
^C
^ Hang here
```

#php -i
```
...
SSL Version => OpenSSL/1.1.1d
...
openssl

OpenSSL support => enabled
OpenSSL Library Version => OpenSSL 1.1.1d  10 Sep 2019
OpenSSL Header Version => OpenSSL 1.1.1  11 Sep 2018
Openssl default config => /usr/lib/ssl/openssl.cnf
...
```

#echo QUIT | openssl s_client -starttls smtp -crlf -connect smtp.mycompany.com:25
```
CONNECTED(00000003)
^C
^ Hang here
```



#echo QUIT | openssl s_client -tls1_2 -crlf -connect smtp.mycompany.com:25
```
CONNECTED(00000003)
write:errno=104
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 221 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1605011544
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
---
```

**Comment by Synchro** [comment]
This combination is very unusual:

    $mail->SMTPSecure = PHPMailer::ENCRYPTION_SMTPS;
    $mail->Port = 25;

This is saying that you want to use implicit TLS on port 25. That is very unlikely to work as it means that connections not using TLS cannot connect to that port. If you're using SMTPS, it would normally be on port 465. You can use `$mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;` on port 25 or 587, but SMTPS does not have that flexibilty.

**Comment by dnsBlah** [comment]
Hi @Synchro ,
Thanks again for your time.
I agree, port 25 for encrypted smtp is unusual, but it's setup like that by our gobal team.
STARTTLS doesn't respond, actually it resets the connection. If you open the connection with TLS1.2 then it'll just say that TLS is already set up.

I think I'm simply missing something on my Ubuntu server setup, as the script works flawlessly on my macBook. I've seen several posts on same like issue, but no resolution. Also I upgraded my php to 7.4 but that doesn't
make a difference. I have no updates for OpenSSL on my Ubuntu server (18.0.4) and been searching how to install LibreSSL, but not really trusting it. Thereby comes I need to compil me nginx from source including the LibreSSL etc..
I seen you have a Ubuntu setup as well in one of the posts, maybe you have some idea's or troubleshooting, updates I can try?

I'd love to get it to work again on my server for automated notifications from my monitoring, reporting etc.

**Comment by Synchro** [comment]
nginx has no bearing on this – PHPMailer is talking to a remote mail server, not the web server you're hosting it on. If you're getting timeouts, It seems possible that your hosting provider may be blocking your outbound connections, but in a way that causes them to hang rather than be rejected immediately, so double check that.

**Comment by dnsBlah** [comment]
Hi @Synchro ,

I'm afraid this is going to be a server issue, as from a remote shell I can simply telnet or netcat etc to the smtp.
However from PHP it just refuses and ends up in a 504 Gateway Timeout error.
My best guess is that the network interface used by PHP is incorrect, unfortunally I don't have enough knowledge on this. Do you have any advice for me?

**Comment by Synchro** [comment]
Given that this fails:

```sh
openssl s_client -connect smtp.mycompany.com:25 -tls1_2
CONNECTED(00000005)
write:errno=104
```
This is the same behaviour your'e seeing inside PHP, so I don't think it's a PHP problem. You say that telnet succeeds from your server – but what do you mean by succeeds? Do you get a welcome banner from your SMTP server, or does it just "connect' and do nothing?

I see mentions of GoDaddy; that's always a bad sign when it comes to email... Is your server hosted with them?

**Comment by dnsBlah** [comment]
:-D
Saying that telnet succeeds, I'm just saying that I can make a connection to the server on the port. So there is no firewall in between being a pain.
Telnet does connect, but once I give any instruction it'll disconnect me because I'm not connection over TLS1.2

The server is hosted on a AWS environment and I guess that the global team has a wildcard certificate with GoDaddy.

But the weird thing is, from my macOS built-in LibreSSL 2.8.3 version it works fine, also from PHP. But from the Ubuntu server using OpenSSL 1.1.1 and LibreSSL 3.2.2 it doesn't. It just resets the connection (server).

I'm simply bouncing on a OpenSSL issue rather than PHP or Nginx.. But unable to solve this.

**Comment by Synchro** [comment]
I'm saying that firewalls can block things in different ways, and so far you've not shown anything that differentiates your telnet responses from your openssl ones. The difference is *not* merely your Mac vs Ubuntu, or libressl vs openssl, but that those machines are on completely different networks. AWS is known for blocking outbound SMTP by default – they want you to use their message service. As I said before, if you want to inspect the TLS config further (if it's even etting that far), use testssl.sh.

**Comment by dnsBlah** [comment]
I'm getting worries about the fact I can connect using telnet, but not with openssl.
I don't see anything on our firewall. And I mist admit, when I'm on the office I'm on the same subnet as my ubuntu servers however I'm on AnyConnect VPN from homeoffice.. *facepalm*
As said the firewall on the datacenter side might not allow me to connect using >TLS< from the servers..
A colleague setup the mailserver on few followme printers, but he can't test it from the webgui... he claims it works from other locations.
So I will go to the office soon, to test it and then I can test my macbook from our IT network.. and being afraid that even my macbook won't work anymore.. well.. i'd be happy if it doesn't anymore. Then I need to findsome support overseas to investigate and have things adjusted on serverside.

I'll update her on the outcome the coming days

**Comment by Synchro** [comment]
> I'm getting worries about the fact I can connect using telnet, but not with openssl.

I don't believe you're seeing a significant difference. telnet is just a dumb client that starts a TCP connection and then does nothing; openssl does TCP (which you don't see), followed by a TLS handshake, so I suspect you're just seeing different failure modes caused by the same thing. I've asked 3 times now for you to run testssl.sh. It will tell you more than either of them.

**Comment by dnsBlah** [comment]
Dear @Synchro 
My apologies for the late response and lack of providing testssl.sh results. Main reason for missing testssl.sh details is that I simply never used it.
I just been on the office today, and it seems to be a firewalling or whitelisting issue on the server side. As I can't connect from the office as well.
As I wanted to make 100% that it's not on my side (network) I provided myself a full open connection and still was unable to connect. Trying to find technical support team from the smtp server now to look into it.
I can connect, but not start the TLS handshake, as the server disconnects me.

I guess this whole ticket was a waste of everyone's time, specially yours where you been friendly and patiently enough to troubleshoot.

Should I close this issue? I will provide an outcome when I finally found the correct team.

Thanks again!
